src/vsphere_cpi/lib/cloud/vsphere/storage_list.rb

```diff
@@ -0,0 +1,28 @@
+module VSphereCloud
+  module StorageList
+
+    # storageList is a list of datastores and datastore_clusters defined in cloud_properties as 'datastores'
+    # Eg. [datastore1, clusters: [{datastore_cluster1: {}, datatore_cluster2: {}}]]
+
+    #Returns list of datastore names
+    def datastore_names
+      storage_list.select do |entry|
+        entry.is_a?(String)
```

Comment:
Let's be friendly to Ruby's duck-typiness. Should be:

```ruby
storage_list.map { |entry| String.try_convert(entry) }.compact
```

See https://idiosyncratic-ruby.com/54-try-converting.html